pkg/build/graph/edges.go

```diff
@@ -40,6 +40,9 @@ const (
 func AddBuildEdges(g osgraph.MutableUniqueGraph, node *buildgraph.BuildConfigNode) {
 	for _, n := range g.(graph.Graph).Nodes() {
 		if buildNode, ok := n.(*buildgraph.BuildNode); ok {
+			if buildNode.Build.Namespace != node.BuildConfig.Namespace {
```

Comment:
I have other things for steve to do in the short term :)  But yes, in the long run.